pyresample/test/test_geometry.py

```diff
@@ -3023,6 +3023,238 @@ def _is_clockwise(lons, lats):
+            "width": 2,
+            "height": 2,
+            "area_extent": (-5326849.0625, -5326849.0625, 5326849.0625, 5326849.0625),
+        }
+        # Define AreaDefintion and retrieve AreaBoundary
+        areadef = geometry.AreaDefinition(**proj_dict_polar_nh)
+        boundary = areadef.boundary(force_clockwise=False)
+
+        # Check boundary shape
+        height, width = areadef.shape
+        n_vertices = (width - 1) * 2 + (height - 1) * 2
+        assert boundary.vertices.shape == (n_vertices, 2)
+
+        # Check boundary vertices is in correct order
+        expected_vertices = np.array([[-135., 55.61313895],
+                                      [135., 55.61313895],
+                                      [45., 55.61313895],
+                                      [-45., 55.61313895]])
+        assert np.allclose(expected_vertices, boundary.vertices)
+
+    def test_geostationary_projection(self):
+        """Test boundary for geostationary projection."""
+        # Define geostationary projection
+        proj_dict_geostationary = {
+            'proj_id': "dummy_geo_projection",
+            "area_id": 'dummy_geo_projection',
+            "description": 'geostationary projection',
+            "projection": {'a': 6378169.00, 'b': 6356583.80, 'h': 35785831.00, 'lon_0': 0, 'proj': 'geos'},
+            "area_extent": (-5500000., -5500000., 5500000., 5500000.),
+            "width": 100,
+            "height": 100,
+        }
+        # Define AreaDefintion and retrieve AreaBoundary
+        areadef = geometry.AreaDefinition(**proj_dict_geostationary)
+
+        # Check default boundary shape
+        default_n_vertices = 50
+        boundary = areadef.boundary(frequency=None)
+        assert boundary.vertices.shape == (default_n_vertices, 2)
+
+        # Check minimum boundary vertices
+        n_vertices = 3
+        minimum_n_vertices = 4
+        boundary = areadef.boundary(frequency=n_vertices)
+        assert boundary.vertices.shape == (minimum_n_vertices, 2)
+
+        # Check odd frequency number
+        # - Rounded to the sequent even number (to construct the sides)
+        n_odd_vertices = 5
+        boundary = areadef.boundary(frequency=n_odd_vertices)
+        assert boundary.vertices.shape == (n_odd_vertices + 1, 2)
+
+        # Check boundary vertices
+        n_vertices = 10
+        boundary = areadef.boundary(frequency=n_vertices, force_clockwise=False)
+        boundary.vertices.shape
+
+        # Check boundary vertices is in correct order
+        expected_vertices = np.array([[-7.92337283e+01, 6.94302533e-15],
+                                      [-7.32756984e+01, 4.40228215e+01],
+                                      [1.81006860e-14, 7.91481122e+01],
+                                      [7.32756984e+01, 4.40228215e+01],
+                                      [7.92337283e+01, -0.00000000e+00],
+                                      [7.92337283e+01, -6.94302533e-15],  # duplicate ?
```

Comment:
This test must be updated when https://github.com/pytroll/pyresample/pull/475 is merged.